Networking/Networking/Model/Refund/Refund.swift

```diff
@@ -0,0 +1,156 @@
+    /// When false, the manual refund process is used.
+    ///
+    public let createAutomatedRefund: Bool?
+
+    public let items: [OrderItemRefund]
+
+    /// Refund struct initializer
+    ///
+    public init(refundID: Int,
+                orderID: Int,
+                siteID: Int,
+                dateCreated: Date,
+                amount: String,
+                reason: String,
+                refundedByUserID: Int,
+                isAutomatedRefund: Bool?,
+                createAutomatedRefund: Bool?,
+                items: [OrderItemRefund]) {
+        self.refundID = refundID
+        self.orderID = orderID
+        self.siteID = siteID
+        self.dateCreated = dateCreated
+        self.amount = amount
+        self.reason = reason
+        self.refundedByUserID = refundedByUserID
+        self.isAutomatedRefund = isAutomatedRefund
+        self.createAutomatedRefund = createAutomatedRefund
+        self.items = items
+    }
+
+    // The public initializer for a Refund
+    ///
+    public init(from decoder: Decoder) throws {
+        guard let orderID = decoder.userInfo[.orderID] as? Int else {
+            throw RefundDecodingError.missingOrderID
+        }
+
+        guard let siteID = decoder.userInfo[.siteID] as? Int else {
+            throw RefundDecodingError.missingSiteID
+        }
+
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let refundID = try container.decode(Int.self, forKey: .refundID)
+        let dateCreated = try container.decodeIfPresent(Date.self, forKey: .dateCreated) ?? Date()
+        let amount = try container.decode(String.self, forKey: .amount)
+        let reason = try container.decode(String.self, forKey: .reason)
+        let refundedByUserID = try container.decode(Int.self, forKey: .refundedByUserID)
+        let isAutomatedRefund = try container.decode(Bool.self, forKey: .automatedRefund)
+        let items = try container.decode([OrderItemRefund].self, forKey: .items)
+
+        self.init(refundID: refundID,
+                  orderID: orderID,
+                  siteID: siteID,
+                  dateCreated: dateCreated,
+                  amount: amount,
+                  reason: reason,
+                  refundedByUserID: refundedByUserID,
+                  isAutomatedRefund: isAutomatedRefund,
+                  createAutomatedRefund: nil,
+                  items: items)
+    }
+
+    // The public initializer for an encodable Refund
+    ///
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(amount, forKey: .amount)
+        try container.encode(reason, forKey: .reason)
+        try container.encode(items, forKey: .items)
+
+        try container.encode(true, forKey: .createAutomatedRefund)
+    }
+}
+
+
+/// Defines all of the Refund CodingKeys
+///
+private extension Refund {
+
+    enum CodingKeys: String, CodingKey {
```

Comment:
Referring back to the previous comment about declaring coding / decoding keys as separate enums, this is one case where I would probably explore declaring two separate sets of coding keys (one for reading and one for writing).

Reply:
done! :)